libs/blocks/merch-card/merch-card.js

```diff
@@ -0,0 +1,216 @@
+    .forEach((cell) => {
+      cell.classList.add(`consonant-${cardType}-cell`);
+    });
+};
+
+const addInner = (el, cardType, merchCard) => {
+  const titles = Array.from(el.querySelectorAll('h1, h2, h3, h4, h5, h6'));
+  const rows = Array.from(el.querySelectorAll('p'));
+  const styles = Array.from(el.classList);
+  const merch = styles.includes('merch-card');
+  const pElement = merch && el.querySelector(':scope > div > div > p:last-of-type');
+  const links = pElement ? pElement.querySelectorAll('a') : el.querySelectorAll('a');
+
+  const inner = el.querySelector(':scope > div:not([class])');
+  inner.classList.add(`consonant-${cardType}-inner`);
+  const title = createTitle(titles, cardType);
+  const description = createDescription(rows, cardType, inner);
+
+  inner.prepend(title);
+  inner.append(description);
+  addFooter(links, inner, merchCard);
+  decorateFooter(el, cardType);
+  merchCard.append(inner);
+};
+
+const decorateRibbon = (el, cardType) => {
+  const ribbonMetadata = el.querySelectorAll('div > div[data-align="center"][data-valign="middle"]');
+
+  if (ribbonMetadata.length >= 2) {
+    const ribbonStyle = ribbonMetadata[0].outerText;
+    const ribbonStyleRegex = /^#[0-9a-fA-F]+, #[0-9a-fA-F]+$/;
+    if (!ribbonStyleRegex.test(ribbonStyle)) return;
+    const [ribbonBgColor, ribbonTextColor] = ribbonStyle.split(', ');
+    const ribbonWrapper = ribbonMetadata[0].parentNode;
+    const ribbon = ribbonMetadata[1];
+    ribbon.classList.add(`consonant-${cardType}-ribbon`);
+    ribbon.style.backgroundColor = ribbonBgColor;
+    ribbon.style.color = ribbonTextColor;
+    el.style.border = `1px solid ${ribbonBgColor}`;
+    const picture = el.querySelector(`.consonant-${cardType}-img`);
+    if (picture) {
+      picture.insertAdjacentElement('afterend', ribbon);
+    } else {
+      el.insertAdjacentElement('beforeend', ribbon);
+    }
+    ribbonWrapper.remove();
+  }
+};
+
+const decorateIcon = (el, icons, cardType) => {
+  if (!icons) return;
+  const inner = el.querySelector(`.consonant-${cardType}-inner`);
+  const iconWrapper = document.createElement('div');
+  iconWrapper.classList.add(`consonant-${cardType}-iconWrapper`);
+  icons?.forEach((icon) => {
+    const url = icon.querySelector('img').src;
+    const iconDiv = document.createElement('div');
+    iconDiv.style.backgroundImage = `url(${url})`;
+    iconDiv.classList.add('consonant-MerchCard-ProductIcon');
+    iconWrapper.appendChild(iconDiv);
+    icon.parentNode?.remove();
+  });
+  inner.prepend(iconWrapper);
+};
+
+const init = (el) => {
+  const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+  decorateLinkAnalytics(el, headings);
+  loadStyle(`${base}/deps/caas.css`);
+  const section = el.closest('.section');
+  section.classList.add('milo-card-section');
+  const images = el.querySelectorAll('picture');
+  let image;
+  const icons = [];
+  let row = el.querySelector(':scope > div');
+  if (row.children.length >= 2) {
```

Comment:
```suggestion
  if (row.children.length >= 2) row = el.querySelectorAll(':scope > *')[1];
```

Reply:
suggestion should be implemented manually

Reply:
done